scripts/test/Muon/results_tab_widget/results_tab_model_test.py

```diff
@@ -0,0 +1,312 @@
+                                    ALLOWED_NON_TIME_SERIES_LOGS):
+            run.addProperty(name,
+                            StringPropertyWithValue(name, 'test'),
+                            replace=True)
+        # verify
+        allowed_logs = log_names(fake_ws.name())
+        for name in itertools.chain(time_series_names,
+                                    ALLOWED_NON_TIME_SERIES_LOGS):
+            self.assertTrue(
+                name in allowed_logs,
+                msg="{} not found in allowed log list".format(name))
+        for name in single_value_log_names:
+            self.assertFalse(name in allowed_logs,
+                             msg="{} found in allowed log list".format(name))
+
+    def test_log_names_from_workspace_without_logs(self):
+        fake_ws = create_test_workspace()
+        allowed_logs = log_names(fake_ws.name())
+        self.assertEqual(0, len(allowed_logs))
+
+    def test_model_returns_fit_functions_from_context(self):
+        self.assertEqual(['func1'], self.model.fit_functions())
+
+    def test_model_returns_no_fit_selection_if_no_fits_present(self):
+        model = ResultsTabModel(FittingContext())
+        self.assertEqual(0, len(model.fit_selection({})))
+
+    def test_model_creates_fit_selection_given_zero_existing_state(self):
+        expected_list_state = {'ws1': [0, True, True]}
+
+        self.assertEqual(expected_list_state, self.model.fit_selection({}))
+
+    def test_model_creates_fit_selection_given_existing_state(self):
+        more_fits = create_test_fits_with_logs(('ws2', ), 'func1',
+                                               self.parameters, self.logs)
+        for fit in more_fits:
+            self.fitting_context.add_fit(fit)
+
+        orig_list_state = {'ws1': [0, False, True]}
+        expected_list_state = {'ws1': [0, False, True], 'ws2': [1, True, True]}
+        self.assertEqual(expected_list_state,
+                         self.model.fit_selection(orig_list_state))
+
+    def test_model_returns_no_log_selection_if_not_fits_present(self):
+        self.fitting_context.fit_list = []
+        self.assertEqual(0, len(self.model.log_selection({})))
+
+    def test_model_returns_log_selection_of_first_workspace(self):
+        self.fitting_context.fit_list = create_test_fits_with_only_workspace_names(
+            ('ws1', 'ws2'))[0]
+        with mock.patch(LOG_NAMES_FUNC) as mock_log_names:
+            ws1_logs = ('run_number', 'run_start')
+            ws2_logs = ('temp', 'magnetic_field')
+
+            def side_effect(name):
+                return ws1_logs if name == 'ws1' else ws2_logs
+
+            mock_log_names.side_effect = side_effect
+            expected_selection = {
+                'run_number': [0, False, True],
+                'run_start': [1, False, True],
+            }
+
+            self.assertEqual(expected_selection, self.model.log_selection({}))
+
+    # def test_model_combines_existing_selection(self):
```

Comment:
should this be here?

Reply:
Thanks. I had commented this out while trying to get something else working and missed re-enabling it.